### PR TITLE
Upgrade Dask version

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR moves us closer to a new `coiled-runtime` release 

Closes https://github.com/coiled/coiled-runtime/issues/100

### CODE REVIEW

**Comment by jrbourbeau** [comment]
Software environment creation seems to be stalled for Python 3.10... I'm going to cancel the build it's it's already been going for over 1.5 hours

**Comment by jrbourbeau** [comment]
Okay, looks like we can't upgrade to the latest `click=8.1.3` release as it's causing us to pull in an older version of `coiled` because `coiled` requires `click>=7.1,<=8.0` to prevent versions of `distributed<2022.04.0` from failing due to https://github.com/dask/distributed/issues/6013.

@dchudz @shughes-uk @ntabris question, thoughts on:

1. Updating the minimum version of `distributed` in `coiled` to be `2022.04.0`? This would remove the need for pinning `click` at all in `coiled`, but my guess is this is too restrictive for all `coiled` users
2. Changing the click pin in `coiled ` to be `click>=7.1,<8.1`. This would allow us to install `click=8.0.4` (which has a Python 3.10 build on `conda-forge`) with `coiled` while continuing to avoid https://github.com/dask/distributed/issues/6013

**Comment by dchudz** [comment]
(1) seems way too restrictive! It's hard to imagine going from "we don't recommend anything later than 2022.01" to "you must use at least 2022.04.0" in one leap.

any downsides to (2)?

**Comment by hayesgb** [comment]
@jrbourbeau @dchudz -- Suggest we wait for 2022.05.1

**Comment by jrbourbeau** [comment]
> (1) seems way too restrictive! 

Yep, thought so but still wanted to jot the option down

> any downsides to (2)?

Not that I'm aware of

> Suggest we wait for 2022.05.1

This PR isn't the actual release, but does add handling around dropping Python 3.7 and adding Python 3.10 with more recent Dask releases. We can always bump the `dask` / `distributed` version again before releasing

**Comment by jrbourbeau** [comment]
See https://github.com/conda-forge/coiled-feedstock/pull/96 for the latest `coiled=0.2.1` release which should resolve the issues we're seeing in CI here by updating the `click` version constraint for `coiled`

**Comment by jrbourbeau** [comment]
@ncclementi this PR should be ready for review -- could you take a look whenever you get a chance?

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -218,6 +212,14 @@ jobs:
           - python-version: "3.9"
             runtime-version: "latest"
             os: "macos-latest"
+          - python-version: "3.7"
```
Comment (ncclementi) [inline .github/workflows/tests.yml:215]: Did you mean 3.8, or we are still testing 3.7 against 0.0.3 only for ubuntu?

Comment (ncclementi) [reply .github/workflows/tests.yml:215]: I see that makes sense, that was my only comment/question. I see CI is green so I'll go ahead and merge this one.

**Pull request merged**